code.js

```diff
@@ -0,0 +1,56 @@
+import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
+
+// Initialize the S3 client outside the handler for reuse
+const s3Client = new S3Client();
+
+/**
+ * Lambda handler for processing orders and storing receipts in S3.
+ * @param {Object} event - Input event containing order details
+ * @param {string} event.order_id - The unique identifier for the order
+ * @param {number} event.amount - The order amount
+ * @param {string} event.item - The item purchased
+ * @returns {Promise<string>} Success message
+ */
+export const handler = async(event) => {
+    try {
+        // Access environment variables
+        const bucketName = process.env.RECEIPT_BUCKET;
+        if (!bucketName) {
+            throw new Error('RECEIPT_BUCKET environment variable is not set');
+        }
+
+        // Create the receipt content and key destination
+        const receiptContent = `OrderID: ${event.order_id}\nAmount: $${event.amount.toFixed(2)}\nItem: ${event.item}`;
+        const key = `receipts/${event.order_id}.txt`;
```

Comment:
> [!WARNING]
> The S3 key is constructed using the raw order_id without validation, which could lead to path traversal attacks if the order_id contains '../' or similar patterns.

> [!TIP]
> Sanitize the order_id to prevent path traversal attacks when constructing the S3 key.

```javascript
// Sanitize order_id for use in the S3 key to prevent path traversal
const sanitizedOrderId = event.order_id.replace(/[^a-zA-Z0-9-_]/g, '');
const key = `receipts/${sanitizedOrderId}.txt`;
```